include/bout/deriv_store.hxx

```diff
@@ -0,0 +1,486 @@
+        theDefault = uppercase(theDefault);
+        defaultMethods[getKey(theDirection, STAGGER::None, theDerivTypeString)] =
+            theDefault;
+        output_info << "The default method for derivative type " << theDerivTypeString;
+        output_info << " in direction " << DIRECTION_STRING(theDirection);
+        output_info << " is " << theDefault << "\n";
+
+        //-------------------------------------------------------------
+        // Staggered
+        //-------------------------------------------------------------
+
+        // The direction specific section to consider with staggering
+        specificSection = options->getSection(direction.second + "stag");
+
+        // Find the appropriate value for theDefault either from
+        // the input file. Note if the specific section for staggering isn't
+        // found then we leave the default as for the non-staggered version
+        if (specificSection->isSet(derivName)) {
+          specificSection->get(derivName, theDefault, "");
+        }
+
+        // Now we have the default method we should store it in defaultMethods
+        theDefault = uppercase(theDefault);
+        defaultMethods[getKey(theDirection, STAGGER::L2C, theDerivTypeString)] =
+            theDefault;
+        defaultMethods[getKey(theDirection, STAGGER::C2L, theDerivTypeString)] =
+            theDefault;
+        output_info << "The default method for staggered derivative type "
+                    << theDerivTypeString;
+        output_info << " in direction " << DIRECTION_STRING(theDirection);
+        output_info << " is " << theDefault << "\n";
+      }
+    }
+  }
+
+  /// The following stores what actual method to use when DIFF_DEFAULT
+  /// is passed. The key is determined using the getKey routine here,
+  /// where the name we pass is determined by the type of method (standard,
+  /// upwind etc.). Note for now we'll always use STAGGER::None as we
+  /// currently assume the default method is independent of staggering --
+  /// it might be useful to relax this assumption!
+  storageType<std::size_t, std::string> defaultMethods;
+
+  /// Provide a method to override/force a specific default method
+  void forceDefaultMethod(std::string methodName, DERIV deriv, DIRECTION direction,
+                          STAGGER stagger = STAGGER::None) {
+    const auto key = getKey(direction, stagger, DERIV_STRING(deriv));
+    defaultMethods[key] = uppercase(methodName);
+  }
+
+  /// Empty all member storage
+  void reset() {
+    defaultMethods.clear();
+    standard.clear();
+    standardSecond.clear();
+    standardFourth.clear();
+    upwind.clear();
+    flux.clear();
+    registeredMethods.clear();
+  }
+
+private:
+  // Make the constructor private so we can't make instances outside
+  // of the struct
+  DerivativeStore(){};
+  // No copy constructor allowed
+  DerivativeStore(const DerivativeStore& junk) = delete;
```

Comment:
These should both be `public` and the default ctor `= delete`

Reply:
If I do this then the line that defines the static instance (`static DerivativeStore instance;`) will not work or am I missing something?

Reply:
Oops, sorry, brain not in gear! I think the deleted copy ctor should be public though